Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -1390,6 +1391,20 @@ def audit_urls
       next unless u =~ %r{https?://(?:central|repo\d+)\.maven\.org/maven2/(.+)$}
       problem "#{u} should be `https://search.maven.org/remotecontent?filepath=#{$1}`"
     end
+
+    return unless @online
+    urls.each do |url|
+      next unless url.start_with? "http:"
+      # Check for insecure mirrors
+      status_code, = curl_output "--connect-timeout", "15", "--output", "/dev/null", "--range", "0-0", \
+                                 "--write-out", "%{http_code}", url
+      secure_url = url.sub "http", "https"
+      secure_status_code, = curl_output "--connect-timeout", "15", "--output", "/dev/null", "--range", "0-0", \
+                                        "--write-out", "%{http_code}", secure_url
+      if status_code.start_with?("20") && secure_status_code.start_with?("20")
+        problem "The URL #{url} could use HTTPS rather than HTTP"
```

Comment:
Great stuff, thanks for picking this up!

To avoid downloading large files, for `url`s I just retrieve the headers and compare `Content-Length:`s returned. If `curl` returned without failure, the size is larger than 1024 bytes and sizes do match, it's fair to assume HTTPS does work. The logic can fall back to full download and content comparison for remaining cases.

As for `homepage`s, it's typical that HTTPS and HTTP flavours return slightly different content, or even that subsequent downloads return slightly different content (due to dynamically generated IDs f.e.). If the size is 512 bytes or larger and contents match, it's a clear case. Otherwise I do these:
- Replace `http://`, `https://`, `http:\/\/`, `https:\/\/` with `//` in both results and then compare them.
- If contents match, it's a clear case match. 
- If only their sizes match, it's a _very likely_ match, where possibly the only difference is some internal IDs.
- If sizes don't match, but the size difference is no larger than +/-10%, I flag them for manual verification, based on "similar size".

For `url`/`appcast`s I found it useful to pass the URL also as `--referrer`. As for the rest of the options, I use these options to get around some other corner-cases: `-fsS -L --max-redirs 10 --user-agent Mozilla/5.0 --cookie-jar ./_cookie --connect-timeout 4`. The values maximums are rather arbitrary though.

I hope this helps.